modules/packages/UnrolledLinkedList.chpl

```diff
@@ -0,0 +1,1249 @@
+      to the right. 
+
+      If the insertion is successful, this method returns `true`. If the given
+      index is out of bounds, this method does nothing and returns `false`.
+
+      .. warning::
+
+        Inserting elements into this unrolledLinkedList may invalidate existing references
+        to the elements contained in this unrolledLinkedList.
+
+      :arg idx: The index into this unrolledLinkedList at which to insert.
+      :type idx: `int`
+
+      :arg arr: An array of elements to insert.
+      :type x: `[] eltType`
+
+      :return: `true` if `arr` was inserted, `false` otherwise.
+      :rtype: `bool`
+    */
+    proc ref insert(idx: int, arr: [?d] eltType): bool lifetime this < arr {
+
+      var result = false;
+
+      on this {
+        _enter();
+        result = _insertGenericKnownSize(idx, arr, arr.size);
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Insert a list of elements `lst` into this unrolledLinkedList at index `idx`, shifting
+      all elements at and following the index `lst.size` positions to the
+      right.
+
+      If the insertion is successful, this method returns `true`. If the given
+      index is out of bounds, this method does nothing and returns `false`.
+
+      .. warning::
+
+        Inserting elements into this unrolledLinkedList may invalidate existing references
+        to the elements contained in this unrolledLinkedList.
+
+      :arg idx: The index into this unrolledLinkedList at which to insert.
+      :type idx: `int`
+
+      :arg lst: A list of elements to insert.
+      :type lst: `list(eltType)`
+
+      :return: `true` if `lst` was inserted, `false` otherwise.
+      :rtype: `bool`
+    */
+    proc ref insert(idx: int, lst: list(eltType)): bool lifetime this < lst {
+      
+      var result = false;
+      
+      // Prevent deadlock if we are trying to insert this into itself.
+      const size = lst.size;
+
+      on this {
+        _enter();
+        result = _insertGenericKnownSize(idx, lst, size);
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Remove the first `count` elements from this unrolledLinkedList with values equal to
+      `x`, shifting all elements following the removed item left.
+
+      If the count of elements to remove is less than or equal to zero, then
+      all elements from this unrolledLinkedList equal to the value of `x` will be removed.
+
+      .. warning::
+
+        Removing elements from this unrolledLinkedList may invalidate existing references
+        to the elements contained in this unrolledLinkedList.
+
+      :arg x: The value of the element to remove.
+      :type x: `eltType`
+
+      :arg count: The number of elements to remove.
+      :type count: `int`
+
+      :return: The number of elements removed.
+      :rtype: `int`
+    */
+    proc ref remove(x: eltType, count:int=1): int {
+      var result = 0;
+
+      on this {
+        _enter(); defer _leave();
```

Comment:
I don't think this is a good practice.

```chapel
_enter();
defer {
  _leave();
}
```

Reply:
Can you explain a little bit why that's better?

Reply:
OK I concede.

There was nothing more to it than splitting lines.